src/ir/expr.cc

```diff
@@ -49,7 +49,7 @@ PrimExpr PrimExpr::FromObject_(ObjectRef ref) {
   if (auto* ptr = ref.as<runtime::StringObj>()) {
     return tir::StringImm(GetRef<runtime::String>(ptr));
   }
-  ICHECK(ObjectTypeChecker<PrimExpr>::Check(ref.get()))
+  ICHECK(!static_cast<bool>(ObjectTypeChecker<PrimExpr>::Mismatch(ref.get())))
       << "Expect type " << ObjectTypeChecker<PrimExpr>::TypeName() << " but get "
       << ref->GetTypeKey();
```

Comment:
ICHECK/CHECK will print out this line, but I don't think it is informative though. We can do this instead. (I wonder if we have an equivalent of `ICHECK` in `LOG(FATAL)`)

```C++
  if (Optional<String> err = ObjectTypeChecker<PrimExpr>::Mismatch(ref.get())) {
    LOG(FATAL) << "Expect type " << ObjectTypeChecker<PrimExpr>::TypeName()
               << " but get " << ref->GetTypeKey() << ". Full error message: " << err.value();
  }
```

Reply:
I'm using `ICHECK(!checked_type.defined())` in other places, so I'll use that here.